### DISCUSSION TITLE
Unexpected behavior from DataLoader on Data with customized attributes

### DISCUSSION THREAD

**Discussion by jipq6175** [pyg-team/pytorch_geometric#4237] [Q&A] [answered]
Hi, first of all, thanks for the amazing lib. 

I encountered an unexpected behavior on `loader.DataLoader` when mini batching the `data.Data` with customized attributes. I have a set of graphs and each graph contains a node mask (tensor.bool). When I use `DataLoader` to create batch data, the node mask was transformed into torch.int. This behavior happens only for certain attribute names. 

Reproduce (pyG version 2.0.3)

```
import networkx as nx

import torch
from torch_geometric.data import Data
from torch_geometric.loader import DataLoader
from torch_geometric.utils import from_networkx

# creating a cyclic graph
g = nx.Graph()
g.add_nodes_from([0, 1, 2])
g.add_edges_from([(0, 1), (1, 2), (2, 0)])

# nx -> pyg data
data = from_networkx(g)

# assign dummy node attributes
data.x = torch.zeros(3, 2)

# different node masking names
data.node_mask = torch.tensor([0, 0, 1], dtype=torch.bool)
data.some_mask = torch.tensor([0, 0, 1], dtype=torch.bool)
data.int_mask = torch.tensor([0, 0, 1], dtype=torch.bool)
data.interfac_mask = torch.tensor([0, 0, 1], dtype=torch.bool)
data.interface_mask = torch.tensor([0, 0, 1], dtype=torch.bool)
data.interface_masks = torch.tensor([0, 0, 1], dtype=torch.bool)
data.face_mask = torch.tensor([0, 0, 1], dtype=torch.bool)

dataloader = DataLoader([data, data, data], batch_size=3, shuffle=False)
for batch_data in dataloader: print(batch_data)
print()

print(batch_data.node_mask)
print(batch_data.some_mask)
print(batch_data.int_mask)
print(batch_data.interfac_mask)
print(batch_data.interface_mask) # ???
print(batch_data.interface_masks) # ???
print(batch_data.face_mask) # ???
```

The output from above is 

```
DataBatch(edge_index=[2, 18], num_nodes=9, x=[9, 2], node_mask=[9], some_mask=[9], int_mask=[9], interfac_mask=[9], interface_mask=[9], interface_masks=[9], face_mask=[9], batch=[9], ptr=[4])

tensor([False, False,  True, False, False,  True, False, False,  True])
tensor([False, False,  True, False, False,  True, False, False,  True])
tensor([False, False,  True, False, False,  True, False, False,  True])
tensor([False, False,  True, False, False,  True, False, False,  True])
tensor([0, 0, 1, 3, 3, 4, 6, 6, 7])
tensor([0, 0, 1, 3, 3, 4, 6, 6, 7])
tensor([0, 0, 1, 3, 3, 4, 6, 6, 7])
``` 

Shouldn't the last three be the same as the others? It seems weird to me some attribute names work and some don't. I wonder what the cause is. Thanks!

**Comment by rusty1s** [comment, marked as answer]
Interesting. By default, we auto-increment any attribute names that contain the word `index` or `face`. For `face`, this assumption is way too strong, and I fixed it in master, see https://github.com/pyg-team/pytorch_geometric/commit/16f236b21546a17f1cacd2099839d0a35130c7d5.

**Reply by jipq6175** [reply]
Thank you
